src/data/actions/grades.js

```diff
@@ -32,15 +35,149 @@ const gradeUpdateFailure = error => ({
 
 
 const toggleGradeFormat = formatType => ({ type: TOGGLE_GRADE_FORMAT, formatType });
+const sortGrades = (columnName, direction) => ({ type: SORT_GRADES, columnName, direction });
 
+const sortAlphaDesc = (gradeRowA, gradeRowB) => {
+  const a = gradeRowA.username.toUpperCase();
+  const b = gradeRowB.username.toUpperCase();
+  if (a < b) {
+    return -1;
+  }
+  if (a > b) {
+    return 1;
+  }
+  return 0;
+};
+
+const sortAlphaAsc = (gradeRowA, gradeRowB) => {
+  const a = gradeRowA.username.toUpperCase();
+  const b = gradeRowB.username.toUpperCase();
+  if (a < b) {
+    return 1;
+  }
+  if (a > b) {
+    return -1;
+  }
+  return 0;
+};
+
+const sortNumerically = (colKey, direction) => {
+  function sortNumAsc(gradeRowA, gradeRowB) {
+    if (gradeRowA[colKey] < gradeRowB[colKey]) {
+      return -1;
+    }
+    if (gradeRowA[colKey] > gradeRowB[colKey]) {
+      return 1;
+    }
+    return 0;
+  }
+
+  function sortNumDesc(gradeRowA, gradeRowB) {
+    if (gradeRowA[colKey] < gradeRowB[colKey]) {
+      return 1;
+    }
+    if (gradeRowA[colKey] > gradeRowB[colKey]) {
+      return -1;
+    }
+    return 0;
+  }
+
+  this.setState({ grades: [...this.state.grades].sort(direction === 'desc' ? sortNumDesc : sortNumAsc) });
+};
+const headingMapper = {
```

Comment:
Move this into a util module so it an be shared by both the grade.js and filter.js